net/httpclient.go

```diff
@@ -320,15 +329,25 @@ func WithBearerToken(t *Transport, bearerToken string) *Transport {
 }
 
 func (t *Transport) shallowCopy() *Transport {
-	tt := *t
-	return &tt
+	return &Transport{
+		once:          sync.Once{},
+		quit:          t.quit,
+		closed:        t.closed,
+		tr:            t.tr,
+		tracer:        t.tracer,
+		spanName:      t.spanName,
+		componentName: t.componentName,
+		bearerToken:   t.bearerToken,
+	}
 }
 
 func (t *Transport) Close() {
-	if !t.closed {
-		t.closed = true
-		close(t.quit)
-	}
+	t.once.Do(func() {
+		if !t.closed {
```

Comment:
If there is no other way to close transport I think we do not need `t.closed` field at all

Reply:
removed t.closed, thanks